source/ui/AMMotorGroupView.cpp

```diff
@@ -17,520 +17,1042 @@ GNU General Public License for more details.
+
+	connect(normalRotationValue_, SIGNAL(editingFinished()),
+			this, SLOT(onNormalRotationValueChanged()));
+
+
+
+}
+
+void AMMotorGroupObjectView::refreshJogUnits()
+{
+	QString jogUnits;
+	QHash<QString, bool> units;
+	if(motorGroupObject_->hasHorizontalAxis()) {
+		if(motorGroupObject_->horizontalAxis()->canRotate()) {
+			units.insert(motorGroupObject_->horizontalAxis()->rotationPositionUnits(), false);
 		}
 
-		else if (viewMode_ == Multiple){
+		if(motorGroupObject_->horizontalAxis()->canTranslate()) {
+			units.insert(motorGroupObject_->horizontalAxis()->translationPositionUnits(), false);
+		}
+	}
 
-			// This is just in case there isn't a selected view yet.
-			if (visibleMotorGroupObjectViews_.isEmpty()){
+	if(motorGroupObject_->hasVerticalAxis()) {
+		if(motorGroupObject_->verticalAxis()->canRotate()) {
+			units.insert(motorGroupObject_->verticalAxis()->rotationPositionUnits(), false);
+		}
 
-				visibleMotorGroupObjectViews_.insert(currentMotorGroupObjectView_->motorGroupObject()->name(), currentMotorGroupObjectView_);
-				currentMotorGroupObjectView_->show();
-			}
+		if(motorGroupObject_->verticalAxis()->canTranslate()) {
+			units.insert(motorGroupObject_->verticalAxis()->translationPositionUnits(), false);
+		}
+	}
+
+	if(motorGroupObject_->hasNormalAxis()) {
+		if(motorGroupObject_->normalAxis()->canRotate()) {
+			units.insert(motorGroupObject_->normalAxis()->rotationPositionUnits(), false);
+		}
 
-			else
-				foreach (AMMotorGroupObjectView *view, visibleMotorGroupObjectViews_.values())
-					view->show();
+		if(motorGroupObject_->normalAxis()->canTranslate()) {
+			units.insert(motorGroupObject_->normalAxis()->translationPositionUnits(), false);
+		}
+	}
 
-			emit motorGroupVisibilityChanged("");
+	foreach (QString currentUnits, units.keys()) {
+		if(jogUnits.isEmpty()) {
+			jogUnits = currentUnits;
+		} else {
+				jogUnits.append(QString("/%1").arg(currentUnits));
 		}
 	}
+
+	jogSize_->setSuffix(QString(" %1").arg(jogUnits));
 }
 
-void AMMotorGroupView::showAvailableMotorGroupChoices(bool show)
+void AMMotorGroupObjectView::errorHandleImplementation(AMMotorGroupObject::MotionDirection,
+													   AMMotorGroupAxis::MotionType,
+													   int)
 {
-	availableMotorGroupObjects_->setVisible(show);
+	// Do nothing here. If subclasses want to do some custom things for the
+	// error case, they can do so in their own override of this function.
 }
 
-void AMMotorGroupView::buildStandardMenuItems(QMenu *menu)
+////////////////////////////////////////////////////////////////////////////////
+// Motor Group View
+////////////////////////////////////////////////////////////////////////////////
+
+AMMotorGroupView::AMMotorGroupView(AMMotorGroup* motorGroup,
+								   ViewMode viewMode,
+								   QWidget* parent)
+	: QWidget(parent)
 {
-	if (viewMode_ == Exclusive){
+	motorGroup_ = motorGroup;
 
-		menu->addAction("Switch to Multiple View");
+	if(!motorGroup_) {
```

Comment:
Aesthetically, I'm not a big fan of returns in constructors.  I think it would be "better" if it was:

```
if (motorGroup_){
     ...
}
```